server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -576,8 +583,33 @@ public Mapping parseMapping(String mappingType, Map<String, Object> mappingSourc
         }
     }
 
-    public static Mapping mergeMappings(DocumentMapper currentMapper, Mapping incomingMapping, MergeReason reason) {
-        return mergeMappings(currentMapper, incomingMapping, reason, Long.MAX_VALUE);
+    public Mapping mergeMappings(Mapping incomingMapping, MergeReason reason) {
```

Comment:
This could stay static as long as all the needed info is provided as argument? I thought I left this comment before but I can on longer find it. it needs the index settings? The calculation could also be done before calling merge? ideally, merge itself remains static.